fields/types/cloudinaryimage/CloudinaryImageType.js

```diff
@@ -31,10 +40,17 @@ function getEmptyValue () {
 function cloudinaryimage (list, path, options) {
 	this._underscoreMethods = ['format'];
 	this._fixedSize = 'full';
-	this._properties = ['select', 'selectPrefix', 'autoCleanup', 'publicID', 'folder', 'filenameAsPublicID'];
+	this._properties = ['select', 'selectPrefix', 'autoCleanup', 'publicID', 'folder', 'filenameAsPublicID', 'generateFilename', 'whenExists', 'retryAttempts'];
 
 	// TODO: implement filtering, usage disabled for now
 	options.nofilter = true;
+	if (options.filenameAsPublicID) {
+		// Produces the same result as the legacy filenameAsPublicID option
+		options.generateFilename = nameFunctions.originalFilename;
+		options.whenExists = 'overwrite';
+	}
+	options = assign({}, DEFAULT_OPTIONS, options);
+	options.generateFilename = ensureCallback(options.generateFilename);
```

Comment:
`options.filenameAsPublicId` support is nicely done :)

Should we default to an undefined filename rather than random though? (leaving cloudinary to generate the public ID as per the old implementation)